src/realm/object-store/sync/impl/sync_client.hpp

```diff
@@ -37,11 +37,11 @@ namespace realm {
 namespace _impl {
 
 struct SyncClient {
-    SyncClient(std::unique_ptr<util::Logger> logger, SyncClientConfig const& config,
+    SyncClient(const std::shared_ptr<util::Logger>& logger, SyncClientConfig const& config,
```

Comment:
Is there any impact on passing it as const ref? We make a copy of it later anyway.

Reply:
I was thinking it would reduce an extra copy of the shared_ptr by passing as a ref, and the const was because the shared_ptr won't be changed by the call.

Reply:
But the shared_ptr _is_ changed by the call by the act of copying it. I don't think it matters either way tho.

Reply:
It turns out, you can't do this if the shared_ptr is not a const:
```
void myfunc(const std::shared_ptr<util::Loger> = nullptr);
```
I had seen the example of the using const std::shared_ptr from Thomas's audit.mm code.
https://github.com/realm/realm-core/blob/master/src/realm/object-store/audit.mm#L639-L640